packages/react-core/src/components/ChipGroup/ChipGroup.tsx

```diff
@@ -17,112 +19,163 @@ export interface ChipGroupProps extends React.HTMLProps<HTMLDivElement> {
+  closeBtnAriaLabel?: string;
+  /** Function that is called when clicking on the chip group close button */
+  onClick?: (event: React.MouseEvent) => void;
+  /** Position of the tooltip which is displayed if the category name text is longer */
+  tooltipPosition?: 'auto' | 'top' | 'bottom' | 'left' | 'right';
 }
 
 interface ChipGroupState {
   isOpen: boolean;
+  isTooltipVisible: boolean;
 }
 
 export class ChipGroup extends React.Component<ChipGroupProps, ChipGroupState> {
   constructor(props: ChipGroupProps) {
     super(props);
     this.state = {
-      isOpen: this.props.defaultIsOpen
+      isOpen: this.props.defaultIsOpen,
+      isTooltipVisible: false
     };
   }
+  private headingRef = React.createRef<HTMLSpanElement>();
 
   static defaultProps: ChipGroupProps = {
-    className: '',
     expandedText: 'Show Less',
     collapsedText: '${remaining} more',
-    withToolbar: false,
+    categoryName: '',
     defaultIsOpen: false,
-    numChips: 3
+    numChips: 3,
+    isClosable: false,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    onClick: (_e: React.MouseEvent) => undefined as any,
+    closeBtnAriaLabel: 'Close chip group',
+    tooltipPosition: 'top',
+    'aria-label': 'Chip group category'
   };
 
+  componentDidMount() {
+    this.setState({
+      isTooltipVisible: Boolean(
+        this.headingRef.current && this.headingRef.current.offsetWidth < this.headingRef.current.scrollWidth
+      )
+    });
+  }
+
   toggleCollapse = () => {
     this.setState(prevState => ({
-      isOpen: !prevState.isOpen
+      isOpen: !prevState.isOpen,
+      isTooltipVisible: Boolean(
+        this.headingRef.current && this.headingRef.current.offsetWidth < this.headingRef.current.scrollWidth
+      )
     }));
   };
 
-  renderToolbarGroup() {
-    const { isOpen } = this.state;
-    const { headingLevel = 'h4' } = this.props;
-    return (
-      <ChipGroupContext.Provider value={headingLevel}>
-        <InnerChipGroup {...this.props} isOpen={isOpen} onToggleCollapse={this.toggleCollapse} />
-      </ChipGroupContext.Provider>
+  renderLabel(id: string) {
+    const { categoryName, tooltipPosition } = this.props;
+    const { isTooltipVisible } = this.state;
+    return isTooltipVisible ? (
+      <Tooltip position={tooltipPosition} content={categoryName}>
+        <span tabIndex={0} ref={this.headingRef} className={css(styles.chipGroupLabel)} aria-hidden="true" id={id}>
+          {categoryName}
+        </span>
+      </Tooltip>
+    ) : (
+      <span ref={this.headingRef} className={css(styles.chipGroupLabel)} aria-hidden="true" id={id}>
+        {categoryName}
+      </span>
     );
   }
 
-  renderChipGroup() {
-    const { className } = this.props;
+  render() {
+    const {
+      categoryName,
+      children,
+      className,
+      isClosable,
+      closeBtnAriaLabel,
+      'aria-label': ariaLabel,
+      onClick,
+      numChips,
+      expandedText,
+      collapsedText,
```

Comment:
Add props to avoid passing to DOM
```ts
      // eslint-disable @typescript-eslint/no-unused-vars
      defaultIsOpen,
      tooltipPosition,
      // eslint-enable @typescript-eslint/no-unused-vars
```